client/extensions/zoninator/state/data-layer/feeds/test/index.js

```diff
@@ -87,3 +97,98 @@ describe( '#updateZoneFeed()', () => {
 		expect( dispatch ).to.have.been.calledWith( updateFeed( 123, 456, fromApi( apiResponse ) ) );
 	} );
 } );
+
+describe( '#saveZoneFeed()', () => {
+	it( 'should dispatch a HTTP request to the feed endpoint', () => {
+		const dispatch = sinon.spy();
+
+		saveZoneFeed( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( http( {
+			method: 'PUT',
+			path: '/jetpack-blogs/123/rest-api/',
+			query: {
+				body: JSON.stringify( toApi( dummyAction.postIds ) ),
+				json: true,
+				path: '/zoninator/v1/zones/456/posts',
+			},
+		}, dummyAction ) );
+	} );
+
+	it( 'should dispatch `removeNotice`', () => {
+		const dispatch = sinon.spy();
+
+		saveZoneFeed( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( removeNotice( 'zoninator-save-feed' ) );
+	} );
+
+	it( 'should dispatch `startSubmit`', () => {
+		const dispatch = sinon.spy();
+
+		saveZoneFeed( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( startSubmit( dummyAction.form ) );
+	} );
+} );
+
+describe( '#announceSuccess()', () => {
+	it( 'should dispatch `stopSubmit`', () => {
+		const dispatch = sinon.spy();
+
+		announceSuccess( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( stopSubmit( dummyAction.form ) );
+	} );
+
+	it( 'should dispatch `initialize`', () => {
+		const dispatch = sinon.spy();
+
+		announceSuccess( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( initialize(
+			dummyAction.form,
+			dummyAction.postIds,
+		) );
+	} );
+
+	it( 'should dispatch `successNotice`', () => {
+		const dispatch = sinon.spy();
+
+		announceSuccess( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( successNotice(
+			translate( 'Zone feed saved!' ),
+			{ id: 'zoninator-save-feed' },
+		) );
+	} );
+
+	it( 'should dispatch `updateFeed`', () => {
+		const dispatch = sinon.spy();
+
+		announceSuccess( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( updateFeed( 123, 456, dummyAction.postIds ) );
+	} );
+} );
+
+describe( '#announceFailure()', () => {
+	it( 'should dispatch `stopSubmit`', () => {
+		const dispatch = sinon.spy();
+
+		announceFailure( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( stopSubmit( dummyAction.form ) );
+	} );
+
+	it( 'should dispatch `errorNotice`', () => {
+		const dispatch = sinon.spy();
+
+		announceFailure( { dispatch }, dummyAction );
+
+		expect( dispatch ).to.have.been.calledWith( errorNotice(
+			translate( 'There was a problem saving your changes. Please try again' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'There was a problem saving your changes. Please try again.' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/feeds-data-layer)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).